### PR TITLE
Fixes delete --all flag for not pushed components

### LINKED ISSUE #1925: odo delete fails for not pushed component
[kind/bug]

## What versions of software are you using?
- Operating System: Win10Pro
- Output of `odo version`:

```
odo v1.0.0-beta3 (c2b402e3)

Server: https://172.17.24.92:8443
Kubernetes: v1.11.0+d4cacc0
```

## How did you run odo exactly?

First, component was created but not pushed:
`odo create java:8 sbcomp`

then 

```
$ odo delete
 X  failed to delete component sbcomp as it doesn't exist
```

## Actual behavior

```
$ odo delete
 X  failed to delete component sbcomp as it doesn't exist
```

## Expected behavior

```
$ odo delete
 X  Component sbcomp deleted
```

### PR DESCRIPTION
**Pull request by mik-dass**
fixes #1925 

How to test
1. Create a component and push it to the cluster
2. Delete it from the cluster without the `all` flag
3. Delete again with the `all` flag and check it deletes the local config file or not

### CODE REVIEW

File: tests/integration/component.go [hunk 1]
```diff
@@ -340,6 +340,14 @@ func componentTests(args ...string) {
 			helper.CmdShouldPass("odo", append(args, "delete", "--all", "-f")...)
 
 		})
+
+		It("creates a local nodejs component without pushing and then deletes it using --all flag", func() {
+			helper.CopyExample(filepath.Join("source", "nodejs"), context)
```
Comment (amitkrout) [inline tests/integration/component.go]:
```
$ odo catalog list components
NAME        PROJECT       TAGS
dotnet      openshift     2.0,latest
httpd       openshift     2.4,latest
nginx       openshift     1.10,1.12,1.8,latest
nodejs      openshift     10,6,8,8-RHOAR,latest
perl        openshift     5.24,5.26,latest
php         openshift     7.0,7.1,latest
python      openshift     2.7,3.5,3.6,latest
ruby        openshift     2.3,2.4,2.5,latest
wildfly     openshift     10.0,10.1,11.0,12.0,13.0,8.1,9.0,latest
```
We have many components in the catalog list. Can you please add some random component type like ```ruby, php, ...``` to add some variety to our test scenario. Moreover this also helps to verify the different default component shown in the catalog list.

File: tests/integration/component.go [hunk 2]
```diff
@@ -340,6 +340,14 @@ func componentTests(args ...string) {
...
+			helper.CmdShouldPass("odo", append(args, "create", "nodejs", "--project", project, "--context", context)...)
+			helper.CmdShouldPass("odo", append(args, "delete", "--all", "-f", "--context", context)...)
+			files := helper.ListFilesInDir(context)
+			Expect(files).NotTo(ContainElement(".odo"))
```
Comment (amitkrout) [inline tests/integration/component.go]: You can add one more scenario like ```creates a local nodejs component without pushing and then deletes it using --all flag``` or you can also append these steps here
```
create component
push component
delete component using ```--all``` flag
Verify the component name should not be present in ```odo component list```
Verify the removal of component dir 
```

Comment (mik-dass) [reply tests/integration/component.go]: I think pushing the component and deleting will add more time for the tests to complete. Also there is already a test to test the `--all` flag.

Comment (amitkrout) [reply tests/integration/component.go]: > I think pushing the component and deleting will add more time for the tests to complete

@mik-dass After pushing the component and testing the --all flag is an important test scenario. We cannot scarifies the scenario in the name of time here atleast ;)

Comment (mik-dass) [reply tests/integration/component.go]: > @mik-dass After pushing the component and testing the --all flag is an important test scenario. We cannot scarifies the scenario in the name of time here atleast ;)

Done

File: tests/integration/component.go [hunk 3]
```diff
@@ -338,7 +338,18 @@ func componentTests(args ...string) {
 			helper.CmdShouldPass("odo", append(args, "create", "nodejs", "--project", project, "--env", "key=value,key1=value1")...)
 			helper.CmdShouldPass("odo", append(args, "push")...)
 			helper.CmdShouldPass("odo", append(args, "delete", "--all", "-f")...)
+			files := helper.ListFilesInDir(context)
+			Expect(files).NotTo(ContainElement(".odo"))
+		})
 
+		It("creates a local nodejs component without pushing and then deletes it using --all flag", func() {
```
Comment (amitkrout) [inline tests/integration/component.go]: "creates a local nodejs component **without pushing** and then deletes it using --all flag" -> "creates a local python component and push then deletes it using --all flag"

File: tests/integration/component.go [hunk 4]
```diff
@@ -338,7 +338,18 @@ func componentTests(args ...string) {
...
+			helper.CopyExample(filepath.Join("source", "pyhton"), context)
+			helper.CmdShouldPass("odo", append(args, "create", "python", "--project", project, "--context", context)...)
+			helper.CmdShouldPass("odo", "push", "--context", context)
+			helper.CmdShouldPass("odo", "delete", "--context", context, "-f")
```
Comment (amitkrout) [inline tests/integration/component.go]: Component name is missing

Comment (amitkrout) [reply tests/integration/component.go]: helper.CmdShouldPass("odo", "delete", "--context", context, "-f") -> helper.CmdShouldPass("odo", append(args, "delete", "compNmae", "--context", context, "-f")...)

File: tests/integration/component.go [hunk 5]
```diff
@@ -338,7 +338,18 @@ func componentTests(args ...string) {
 			helper.CmdShouldPass("odo", append(args, "create", "nodejs", "--project", project, "--env", "key=value,key1=value1")...)
 			helper.CmdShouldPass("odo", append(args, "push")...)
 			helper.CmdShouldPass("odo", append(args, "delete", "--all", "-f")...)
+			files := helper.ListFilesInDir(context)
```
Comment (amitkrout) [inline tests/integration/component.go]: +1

File: tests/integration/component.go (hunk 4 above)
Comment (amitkrout) [reply tests/integration/component.go]: Also just add one more validation step here as i mentioned in one of my comment
```
Verify the component name should not be present in ```odo component list```
```

File: tests/integration/component.go [hunk 6]
```diff
@@ -335,10 +338,25 @@ func componentTests(args ...string) {
 
 		It("creates and pushes local nodejs component and then deletes --all", func() {
 			helper.CopyExample(filepath.Join("source", "nodejs"), context)
-			helper.CmdShouldPass("odo", append(args, "create", "nodejs", "--project", project, "--env", "key=value,key1=value1")...)
-			helper.CmdShouldPass("odo", append(args, "push")...)
-			helper.CmdShouldPass("odo", append(args, "delete", "--all", "-f")...)
-
+			helper.CmdShouldPass("odo", append(args, "create", "nodejs", componentName, "--app", appName, "--project", project, "--env", "key=value,key1=value1")...)
+			helper.CmdShouldPass("odo", append(args, "push", "--context", context)...)
+			helper.CmdShouldPass("odo", append(args, "delete", "--context", context, "-f", "--all", "--app")...)
```
Comment (amitkrout) [inline tests/integration/component.go]:
```
Running odo with args [odo delete --context /tmp/286224696 -f --all --app]
[odo] Error: flag needs an argument: --app
```

Comment (amitkrout) [reply tests/integration/component.go]: Causes CI failure

Comment (mik-dass) [reply tests/integration/component.go]: @amitkrout Fixed

**Pull request merged**